File: src/main/java/ma/fstg/gestionprojets/services/PhaseService.java
package ma.fstg.gestionprojets.services;
import ma.fstg.gestionprojets.dto.request.PhaseRequest;
import ma.fstg.gestionprojets.dto.response.PhaseResponse;
import ma.fstg.gestionprojets.entities.Phase;
import java.util.List;
public interface PhaseService {
    PhaseResponse create(Long projetId, PhaseRequest req);
    PhaseResponse update(Long id, PhaseRequest req);
    PhaseResponse getById(Long id);
    List<PhaseResponse> getByProjet(Long projetId);
    List<PhaseResponse> getMyPhases(String login);
    List<PhaseResponse> getAll();
    PhaseResponse updateRealisation(Long id, boolean etat);
    PhaseResponse updateFacturation(Long id, boolean etat);
    PhaseResponse updatePaiement(Long id, boolean etat);
    void delete(Long id);
    Phase findEntityById(Long id);
    PhaseResponse toResponse(Phase p);
}

File: gestion-projets-frontend/src/services/phaseService.ts
import api from '../utils/axiosConfig';

export const phaseService = {
  getByProjet: (projetId: number) => api.get(`/projets/${projetId}/phases`),
  getById: (id: number) => api.get(`/phases/${id}`),
  create: (projetId: number, data: any) => api.post(`/projets/${projetId}/phases`, data),
  update: (id: number, data: any) => api.put(`/phases/${id}`, data),
  delete: (id: number) => api.delete(`/phases/${id}`),
  updateRealisation: (id: number, etat: boolean) => api.patch(`/phases/${id}/realisation`, { etat }),
  updateFacturation: (id: number, etat: boolean) => api.patch(`/phases/${id}/facturation`, { etat }),
  updatePaiement: (id: number, etat: boolean) => api.patch(`/phases/${id}/paiement`, { etat }),
};
